snapshot-cassandra/src/main/scala/com/evolutiongaming/kafka/journal/snapshot/cassandra/SnapshotStatements.scala

```diff
@@ -0,0 +1,309 @@
+package com.evolutiongaming.kafka.journal.snapshot.cassandra
+
+import cats.Monad
+import cats.syntax.all._
+import com.datastax.driver.core.Row
+import com.evolutiongaming.kafka.journal._
+import com.evolutiongaming.kafka.journal.cassandra.CassandraConsistencyConfig
+import com.evolutiongaming.kafka.journal.eventual.EventualPayloadAndType
+import com.evolutiongaming.kafka.journal.eventual.cassandra.CassandraHelper._
+import com.evolutiongaming.kafka.journal.eventual.cassandra.CassandraSession
+import com.evolutiongaming.scassandra.syntax._
+import com.evolutiongaming.scassandra.{DecodeByName, EncodeByName, TableName}
+import scodec.bits.ByteVector
+
+import java.time.Instant
+
+object SnapshotStatements {
+
+  def createTable(name: TableName): String = {
+    s"""
+       |CREATE TABLE IF NOT EXISTS ${name.toCql} (
+       |id TEXT,
+       |topic TEXT,
+       |buffer_idx INT,
+       |seq_nr BIGINT,
+       |timestamp TIMESTAMP,
+       |origin TEXT,
+       |version TEXT,
+       |metadata TEXT,
+       |payload_type TEXT,
+       |payload_txt TEXT,
+       |payload_bin BLOB,
+       |PRIMARY KEY ((id, topic), buffer_idx))
+       |""".stripMargin
+  }
+
+  trait InsertRecord[F[_]] {
+    def apply(key: Key, bufferNr: BufferNr, snapshot: SnapshotRecord[EventualPayloadAndType]): F[Boolean]
+  }
+
+  object InsertRecord {
+
+    def of[F[_]: Monad: CassandraSession](
+      name: TableName,
+      consistencyConfig: CassandraConsistencyConfig.Write
+    ): F[InsertRecord[F]] = {
+
+      implicit val encodeByNameByteVector: EncodeByName[ByteVector] =
+        EncodeByName[Array[Byte]].contramap(_.toArray)
+
+      val query =
+        s"""
+           |INSERT INTO ${name.toCql} (
+           |id,
+           |topic,
+           |buffer_idx,
+           |seq_nr,
+           |timestamp,
+           |origin,
+           |version,
+           |payload_type,
+           |payload_txt,
+           |payload_bin,
+           |metadata)
+           |VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)
+           |IF NOT EXISTS
```

Comment:
As was pointed out by @migesok, [mixing LWT and normal operations can result in errors](https://docs.datastax.com/en/cassandra-oss/3.0/cassandra/dml/dmlLtwtTransactions.html):

> Lightweight transactions will block other lightweight transactions from occurring, but will not stop normal read and write operations from occurring. Lightweight transactions use a timestamping mechanism different than for normal operations and mixing LWTs and normal operations can result in errors. If lightweight transactions are used to write to a row within a partition, only lightweight transactions for both read and write operations should be used. This caution applies to all operations, whether individual or batched.

@rtar, could we instead of LWT rely on Single Writer Principle provided by Akka?

Reply:
This paragraph only applies to the write operations and the read operations requiring a strong consistency requirement.

It should be safe to read the data, given you are okay with eventual consistency or if you use SERIAL consistency level while reading it.

The documentation is very confusing in that area.

Saying that, after considering this for a longer time, I decided that LWTs are totally unnecessary here, indeed, given a single writer is guaranteed externally, as you suggested. 

I already started a work to remove LWTs. I thought I shared this with @migesok, but I think I confused him by saying that writing snapshot will not be guaranteed. It will be guaranteed, actually, given the single writer is ensured.

Reply:
Thanks for explanation 🙇 , I was not aware of your plans to move on with SWP instead of LWT thus will be happy to review final version of the snapshot store

Reply:
LWTs are disabled by default, now.